test/aggregate.test.js

```diff
@@ -1242,4 +1242,16 @@ describe('aggregate: ', function() {
     });
   });
 
+  it('should not throw error if database connection has not been established (gh-13125)', async function() {
+    const m = new mongoose.Mongoose();
+    const mySchema = new Schema({ test: String });
+    const M = m.model('Test', mySchema);
+
+    const aggregate = M.aggregate();
+    aggregate.match({ $text: { $search: 'test' } });
+
+    assert.doesNotThrow(async function() {
+      await aggregate.exec();
```

Comment:
I suspect this will cause some issues because this operation will eventually time out because of buffering. Probably best to save the promise and then await on it later, for example `const promise = aggregate.exec(); await m.connect(start.uri); const res = await promise;` similar to https://github.com/Automattic/mongoose/blob/c4fd23d64087510d3293f7168b06c284abf52a3c/test/collection.test.js#L37

Reply:
or like in other parts of the tests, use [`await assert.rejects`](https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message)